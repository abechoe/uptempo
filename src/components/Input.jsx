import Radium from 'radium';
import React from 'react';

class Input extends React.Component {
  constructor(props) {
    super(props);
    this.handleInputChange = this.handleInputChange.bind(this);
  }

  handleInputChange(event) {
    if (this.props.name === 'event_interval_join_attributes') {
      const interval_attrs = {
        interval_attributes: {
          seconds: event.target.value,
        }
      }
      this.props.onChange({[this.props.name]: interval_attrs});
    } else {
      this.props.onChange({[this.props.name]: event.target.value});
    }
  }

  render() {
    return (
      <div>
        <label htmlFor={this.props.id}>{this.props.title}:</label>
        <input type={this.props.type}
          id={this.props.id}
          name={this.props.name}
          value={this.props.value}
          onChange={this.handleInputChange}
          style={styles.base}
        />
      </div>
    );
  }
}

const styles = {
  base: {
    display: 'block',
    margin: '2px 0 10px 0',
    borderRadius: '5px',
    padding: '1px',
    height: '20px',
  }
}

export default Radium(Input);