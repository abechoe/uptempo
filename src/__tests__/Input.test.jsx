import React from 'react';
import { shallow } from 'enzyme';

import Input from 'components/Input';

describe('Input component tests', () => {
  it('renders an Input component', () => {
    const wrapper = shallow(<Input type="text" id="name" name="name" value="Bob" />)
    expect(wrapper).toMatchSnapshot();
  });
})