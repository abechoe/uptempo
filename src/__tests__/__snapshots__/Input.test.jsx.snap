// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input component tests renders an Input component 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Input
    id="name"
    name="name"
    type="text"
    value="Bob"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <label
          data-radium={true}
          htmlFor="name"
        >
          :
        </label>,
        <input
          data-radium={true}
          id="name"
          name="name"
          onChange={[Function]}
          style={
            Object {
              "borderRadius": "5px",
              "display": "block",
              "height": "20px",
              "margin": "2px 0 10px 0",
              "padding": "1px",
            }
          }
          type="text"
          value="Bob"
        />,
      ],
      "data-radium": true,
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": ".0",
        "nodeType": "host",
        "props": Object {
          "children": Array [
            ":",
          ],
          "data-radium": true,
          "htmlFor": "name",
        },
        "ref": null,
        "rendered": Array [
          ":",
        ],
        "type": "label",
      },
      Object {
        "instance": null,
        "key": ".1",
        "nodeType": "host",
        "props": Object {
          "children": undefined,
          "data-radium": true,
          "id": "name",
          "name": "name",
          "onChange": [Function],
          "style": Object {
            "borderRadius": "5px",
            "display": "block",
            "height": "20px",
            "margin": "2px 0 10px 0",
            "padding": "1px",
          },
          "type": "text",
          "value": "Bob",
        },
        "ref": null,
        "rendered": null,
        "type": "input",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <label
            data-radium={true}
            htmlFor="name"
          >
            :
          </label>,
          <input
            data-radium={true}
            id="name"
            name="name"
            onChange={[Function]}
            style={
              Object {
                "borderRadius": "5px",
                "display": "block",
                "height": "20px",
                "margin": "2px 0 10px 0",
                "padding": "1px",
              }
            }
            type="text"
            value="Bob"
          />,
        ],
        "data-radium": true,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": ".0",
          "nodeType": "host",
          "props": Object {
            "children": Array [
              ":",
            ],
            "data-radium": true,
            "htmlFor": "name",
          },
          "ref": null,
          "rendered": Array [
            ":",
          ],
          "type": "label",
        },
        Object {
          "instance": null,
          "key": ".1",
          "nodeType": "host",
          "props": Object {
            "children": undefined,
            "data-radium": true,
            "id": "name",
            "name": "name",
            "onChange": [Function],
            "style": Object {
              "borderRadius": "5px",
              "display": "block",
              "height": "20px",
              "margin": "2px 0 10px 0",
              "padding": "1px",
            },
            "type": "text",
            "value": "Bob",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
